dcc-portal-ui/app/scripts/sets/js/services.js

```diff
@@ -603,4 +603,69 @@
     _service.refreshList();
   });
 
+  module.service('SetNameService', function(LocationService, FilterService, SetService, FiltersUtil, 
+    CompoundsService, GeneSymbols, $filter){
+    
+    var _service = this;
+
+    // Function to get UI friendly filter
+    _service.getSetFilters = function() {
+      var ids = LocationService.extractSetIds(FilterService.filters());
+      return ids.length ? SetService.getMetaData(ids).then(function (results) {
+            return FiltersUtil.buildUIFilters (FilterService.filters(), SetService.lookupTable(results.plain()));
+          }) : Promise.resolve(FiltersUtil.buildUIFilters(FilterService.filters(), {}));
+    }
+
+    // Returns the promise to get the set name
+    _service.getSetName = function(filters, setType) {
+      if (_.isEmpty(filters)) {
+        return Promise.resolve('All ' + _.capitalize(setType) + 's');
+      }
+
+      // Going throught filters to create a custom Collection of filters
+      var promises = _(filters).map(function (filter, filterKey) {
+        return _.map(filter, function (facet, facetKey) {
+          return facet.is.map(function (value) {
+            return { term: value.term, facetName: value.controlFacet, facetGroup: filterKey + facetKey, operator: 'is' };
+          }).concat((facet.not || []).map(function (value) {
+            return { term: value.term, facetName: value.controlFacet, facetGroup: filterKey + facetKey, operator: 'not' };
+          }));
+        });
+      })
+      .flattenDeep()
+      .groupBy('facetGroup')
+      .map(function (facetTermWrappers) {
+        var partialNamesRequests = facetTermWrappers.map(function (termWrapper) {
+          // If filter is compound, need to return CompoundService promise that will get the name based on compound ID
+          if (termWrapper.facetName === 'compoundId') {
+            return CompoundsService.getCompoundByZincId(termWrapper.term).then(function (compound) {
+              return _.capitalize(compound.name);
+            });
+            // Following condition will get the Gene name based on ID
+          } else if (termWrapper.facetName === 'id' && termWrapper.term.indexOf('ENSG') > -1) {
```

Comment:
By itself it doesn't but we have `&&` in that condition which in turn makes sure it is gene id. 
`termWrapper.facetName === 'id' && termWrapper.term.indexOf('ENSG') > -1`